File: src/test/java/com/agiledeveloper/FibonacciTest.java
package com.agiledeveloper;

import org.junit.jupiter.api.Test;

import java.util.Arrays;

import static org.junit.jupiter.api.Assertions.assertAll;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class FibonacciTest {
  @Test
  void numberAtPosition() {
    assertAll(
      () -> assertEquals(1, Fibonacci.numberAtPosition(0))
      , () -> assertEquals(1, Fibonacci.numberAtPosition(1))
      , () -> assertEquals(2, Fibonacci.numberAtPosition(2))
      , () -> assertEquals(3, Fibonacci.numberAtPosition(3))
      , () -> assertEquals(5, Fibonacci.numberAtPosition(4))
      , () -> assertEquals(8, Fibonacci.numberAtPosition(5))
      , () -> assertEquals(89, Fibonacci.numberAtPosition(10))
    );
  }

  @Test
  void series() {
    assertAll(
      () -> assertEquals(Arrays.asList(1), Fibonacci.seriesUntilPosition(0))
      , () -> assertEquals(Arrays.asList(1, 1), Fibonacci.seriesUntilPosition(1))
      , () -> assertEquals(Arrays.asList(1, 1, 2), Fibonacci.seriesUntilPosition(2))
      , () -> assertEquals(Arrays.asList(1, 1, 2, 3), Fibonacci.seriesUntilPosition(3))
      , () -> assertEquals(Arrays.asList(1, 1, 2, 3, 5), Fibonacci.seriesUntilPosition(4))
      , () -> assertEquals(Arrays.asList(1, 1, 2, 3, 5, 8), Fibonacci.seriesUntilPosition(5))
      , () -> assertEquals(Arrays.asList(1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89), Fibonacci.seriesUntilPosition(10))
    );
  }
}

File: src/test/scala/com/agiledeveloper/FibonacciTest.scala
package com.agiledeveloper

import org.scalatest._

class FibonacciTest extends WordSpec with Matchers {

  "Fibonacci" should {
    "compute numbers in the series at a given position" in {
      Fibonacci.numberAtPosition(0) shouldBe 1
      Fibonacci.numberAtPosition(1) shouldBe 1
      Fibonacci.numberAtPosition(2) shouldBe 2
      Fibonacci.numberAtPosition(3) shouldBe 3
      Fibonacci.numberAtPosition(4) shouldBe 5
      Fibonacci.numberAtPosition(5) shouldBe 8
      Fibonacci.numberAtPosition(10) shouldBe 89
    }

    "collect the series up to a given position" in {
      Fibonacci.seriesUntilPosition(0) shouldBe List(1)
      Fibonacci.seriesUntilPosition(1) shouldBe List(1, 1)
      Fibonacci.seriesUntilPosition(2) shouldBe List(1, 1, 2)
      Fibonacci.seriesUntilPosition(3) shouldBe List(1, 1, 2, 3)
      Fibonacci.seriesUntilPosition(4) shouldBe List(1, 1, 2, 3, 5)
      Fibonacci.seriesUntilPosition(5) shouldBe List(1, 1, 2, 3, 5, 8)
      Fibonacci.seriesUntilPosition(10) shouldBe List(1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89)
    }
  }
}

